pbtime/cmp.go

```diff
@@ -0,0 +1,32 @@
+package pbtime
+
+import (
+	"time"
+
+	tspb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func IsZero(t *tspb.Timestamp) bool {
+	return t == nil || t.Nanos == 0 && t.Seconds == 0
+}
```

Comment:
this isn't generically true. it's sufficient if we assume that the time is in the future, but technically 1970-01-01 is a valid timestamp. generically `IsZero` is equivalent to `t == nil`

Reply:
This function doesn't do a validation. The goal is to make a canonical zero value which work with a non nil version of `tspb.Timestamp`.

Reply:
But it's not zero. We can't use this to infer that

Reply:
It's zero value of `tspb.Timestamp` and I thought that yesterday we said to add a support for a non nil zero value. 

That being said, my if  we can change non nil to nil in proto  and handle that in a normal way, then that would be my preference.